synapse/lib/snap.py

```diff
@@ -39,6 +37,8 @@ def __init__(self, core, layers, write=False):
         self.strict = True
         self.elevated = False
 
+        self.permcache = s_cache.FixedCache({}, size=1000)
```

Comment:
Setting `callback` to `{}` is probably the wrong thing to do in the case where these is no user. Is there something better I could do here?